code-studio/src/js/levels/dialogHelper.js

```diff
@@ -200,6 +202,20 @@ window.dashboard.dialog = (function () {
 
         if (lastServerResponse.videoInfo) {
           window.dashboard.videos.showVideoDialog(lastServerResponse.videoInfo);
+        } else if (lastServerResponse.endOfStageExperience) {
+          var body = document.createElement('div');
+          var props = {
+            stageName: `Stage ${lastServerResponse.previousStageInfo.position}: ${lastServerResponse.previousStageInfo.name}`,
+            onContinue: () => { dialog.hide(); },
+            i18n: window.dashboard.i18n
+          };
+          ReactDOM.render(<PlayZone {...props} />, body);
```

Comment:
Another place where I'd probably just pass in the props directly instead of creating a props object.